src/main/java/core/basesyntax/db/StockDaoStorageImpl.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.db;
+
+import core.basesyntax.storage.Storage;
+import java.util.Set;
+
+public class StockDaoStorageImpl implements StockDao {
+
+    @Override
+    public void set(String product, Integer amount) {
+        Storage.stock.put(product, amount);
+    }
+
+    @Override
+    public Integer get(String product) {
+        return Storage.stock.get(product);
+    }
+
+    @Override
+    public void decrease(String product, Integer amount) {
+        if (contain(product)) {
+            Storage.stock.put(product, get(product) - amount);
+        } else if (get(product) - amount < 0) {
+            throw new RuntimeException(
+                    "the quantity being subtracted is greater than the stock value");
+        } else {
+            throw new RuntimeException("There was no such product before");
+        }

```

Comment:
The logic in the `decrease` method is flawed. The check for `get(product) - amount < 0` should be inside the `if (contain(product))` block. Currently, it is incorrectly placed in the `else if` block, which will never be executed because the `else` block will be executed if the product does not exist.